var/spack/repos/builtin/packages/environment-modules/package.py

```diff
@@ -3,64 +3,84 @@
+    version('4.1.2', sha256='d1f54f639d1946aa1d7ae8ae03752f8ac464a879c14bc35e63b6a87b8a0b7522')
+    version('4.1.1', sha256='998e9cc936045b4e84f28ca60c4680c08385a210d6bb95fc31c28a7d634a9357')
+    version('4.1.0', sha256='d98aa369219bf0a4ec41efe7cb8d1127d34cb07666088dd79da6b424196d4cfd')
+    version('4.0.0', sha256='f0ab7f6a747863cb980681a904a3c9380e2e52de6eb046cfa285e5e225f9ac47')
+    version(
+        '3.2.10', sha256='fb05c82a83477805a1d97737a9f0ca0db23f69b7bce504f1609ba99477b03955',
+        url='http://prdownloads.sourceforge.net/modules/modules-3.2.10.tar.gz'
+    )
 
     variant('X', default=True, description='Build with X functionality')
 
     # Dependencies:
     depends_on('tcl', type=('build', 'link', 'run'))
+    depends_on('tcl@8.4:', type=('build', 'link', 'run'), when='@4.0.0:')
 
     def install(self, spec, prefix):
-        tcl_spec = spec['tcl']
-
-        # We are looking for tclConfig.sh
-        tcl_config_name = 'tclConfig.sh'
-        tcl_config_dir_options = [tcl_spec.prefix.lib,
-                                  tcl_spec.prefix.lib64]
-
-        tcl_config_found = False
-        for tcl_config_dir in tcl_config_dir_options:
-            tcl_config_found = os.path.exists(
-                join_path(tcl_config_dir, tcl_config_name))
-            if tcl_config_found:
-                break
-
-        if not tcl_config_found:
-            raise InstallError('Failed to locate ' + tcl_config_name)
+        tcl = spec['tcl']
 
-        # See: https://sourceforge.net/p/modules/bugs/62/
-        cpp_flags = ['-DUSE_INTERP_ERRORLINE']
+        # Determine where we can find tclConfig.sh
+        for tcl_lib_dir in [tcl.prefix.lib, tcl.prefix.lib64]:
+            tcl_config_file = os.path.join(tcl_lib_dir, 'tclConfig.sh')
+            if os.path.exists(tcl_config_file):
+                break
+        else:
+            raise InstallError('Failed to locate tclConfig.sh')
 
         config_args = [
+            "--prefix=" + prefix,
             "--without-tclx",
             "--with-tclx-ver=0.0",
-            "--prefix=" + prefix,
             # It looks for tclConfig.sh
-            "--with-tcl=" + tcl_config_dir,
-            "--with-tcl-ver=%d.%d" % (
-                tcl_spec.version.version[0], tcl_spec.version.version[1]),
-            '--disable-debug',
+            "--with-tcl=" + tcl_lib_dir,
+            "--with-tcl-ver={0}.{1}".format(*tcl.version.version[0:2]),
             '--disable-dependency-tracking',
             '--disable-silent-rules',
             '--disable-versioning',
-            '--datarootdir=' + prefix.share,
-            'CPPFLAGS=' + ' '.join(cpp_flags)
+            '--datarootdir=' + prefix.share
         ]
 
         if '~X' in spec:
             config_args = ['--without-x'] + config_args
 
+        if '@4.2.0:' in self.spec:
+            config_args.extend([
+                '--enable-auto-handling'
+            ])
+
```

Comment:
Starting version 4.1.0, I would suggest to add the following `./configure` argument:

     --with-quarantine-vars=LD_LIBRARY_PATH

This setting enables [Modules quarantine mechanism](https://modules.readthedocs.io/en/stable/MIGRATING.html#quarantine-mechanism-to-protect-module-execution), which avoids being poisoned by an incompatible Tcl installation coming from one software bundle enabled in user environment.

Reply:
Last request: can you double check the comment I wrote in 69d9dd8? Just to be sure my understanding is correct and that I won't put the next guy changing this package on a wrong track...